src/lib/flash-helper.js

```diff
@@ -174,11 +175,31 @@ async function createFlashSteps({ modules, isInDfuMode, platformId }) {
 			flashStep.flashMode = 'normal';
 			normalModules.push(flashStep);
 		} else {
+			if (moduleType === 'userPart') {
+				deviceOsVersion = module.prefixInfo.depModuleVersion;
+			}
```

Comment:
Move the block of code below to inside the `if` here. It's very important that you push the invalidate step to `dfuModules` BEFORE you push the flash application step otherwise it might erase part of the application you just flashed.